Client/game_sa/CHandlingManagerSA.cpp

```diff
@@ -192,6 +242,54 @@ const CHandlingEntry* CHandlingManagerSA::GetOriginalHandlingData(eVehicleTypes
     return NULL;
 }
 
+const CFlyingHandlingEntry* CHandlingManagerSA::GetOriginalFlyingHandlingData(eVehicleTypes eModel)
+{
+    // Within range?
+    if (eModel >= 400 && eModel < VT_MAX)
+    {
+        // Get our Handling ID
+        eHandlingTypes eHandling = GetHandlingID(eModel);
+        // Original GTA:SA behavior
+        if (eHandling < 186 || eHandling > 209)
+            return m_pOriginalFlyingEntries[0];
+        else
+            return m_pOriginalFlyingEntries[eHandling - 186];
+    }
+
+    return NULL;
+}
+
+const CBoatHandlingEntry* CHandlingManagerSA::GetOriginalBoatHandlingData(eVehicleTypes eModel)
+{
+    // Within range?
+    if (eModel >= 400 && eModel < VT_MAX)
+    {
+        // Get our Handling ID
+        eHandlingTypes eHandling = GetHandlingID(eModel);
+        // Original GTA:SA behavior
+        if (eHandling < 175 || eHandling > 186)
+            return m_pOriginalBoatEntries[0];
+        else
+            return m_pOriginalBoatEntries[eHandling - 175];
+    }
+
+    return NULL;
+}
+
+const CBikeHandlingEntry* CHandlingManagerSA::GetOriginalBikeHandlingData(eVehicleTypes eModel)
+{
+    // Within range?
+    if (eModel >= 400 && eModel < VT_MAX)
+    {
+        // Get our Handling ID
+        eHandlingTypes eHandling = GetHandlingID(eModel);
+        if (eHandling >= HT_BIKE || eHandling <= HT_FREEWAY)
```

Comment:
I consistently get game crashes (offset 0x0000A6B6) on the latest nightly (r20716) (does not occur on a debug build) whenever a car spawns (specifically tested with a banshee but it happens with other cars too, but not bikes, helis, planes, etc)
The crash is a read access violation on line 28 of CBikeHandlingEntrySA.cpp on `CBikeHandlingEntrySA::Assign` called from `CClientVehicle::CClientVehicle` constructor, only if `CHandlingManagerSA::GetOriginalBikeHandlingData` returns a valid pointer which should only happen with handling IDs between `HT_BIKE` and `HT_FREEWAY` but this line wrongly catches pretty much all handling IDs there are.
```suggestion
        if (eHandling >= HT_BIKE && eHandling <= HT_FREEWAY)
```

Reply:
Nice catch!